solr/contrib/ltr/src/java/org/apache/solr/ltr/store/rest/ManagedModelStore.java

```diff
@@ -299,36 +208,95 @@ public String toString() {
-        final Map<String,Object> map = new HashMap<String,Object>(2, 1.0f);
-        map.put("name", feature.getName());
-        map.put("norm", toNormalizerMap(norm));
-        features.add(map);
+  private static List<Object> modelsAsManagedResources(List<LTRScoringModel> models) {
+    final List<Object> list = new ArrayList<>(models.size());
+    for (final LTRScoringModel model : models) {
+      list.add(toLTRScoringModelMap(model));
+    }
+    return list;
+  }
+
+  @SuppressWarnings("unchecked")
+  public static LTRScoringModel fromLTRScoringModelMap(SolrResourceLoader solrResourceLoader,
+      Map<String,Object> modelMap, ManagedFeatureStore managedFeatureStore) {
+
+    final FeatureStore featureStore =
+        managedFeatureStore.getFeatureStore((String) modelMap.get(STORE_KEY));
+
+    final List<Feature> features = new ArrayList<>();
+    final List<Normalizer> norms = new ArrayList<>();
+
+    final List<Object> featureList = (List<Object>) modelMap.get(FEATURES_KEY);
+    if (featureList != null) {
+      for (final Object feature : featureList) {
+        final Map<String,Object> featureMap = (Map<String,Object>) feature;
+        features.add(featureFromFeatureMap(solrResourceLoader, featureMap, featureStore));
+        norms.add(normalizerFromFeatureMap(solrResourceLoader, featureMap));
       }
-      modelMap.put("features", features);
-      modelMap.put("params", modelmeta.getParams());
+    }
 
-      list.add(modelMap);
+    LTRScoringModel model;
+    try {
+      model = LTRScoringModel.getInstance(solrResourceLoader,
+          (String) modelMap.get(CLASS_KEY), // modelClassName
+          (String) modelMap.get(NAME_KEY), // modelName
+          features,
+          norms,
+          featureStore.getName(),
+          featureStore.getFeatures(),
+          (Map<String,Object>) modelMap.get(PARAMS_KEY));
+    } catch (ModelException ex) {
+      log.info("ModelException {}", ex);
+      throw ex;
     }
-    return list;
+
+    return model;
   }
-  
-  private static Normalizer fromNormalizerMap(SolrResourceLoader solrResourceLoader,
+
+  private static LinkedHashMap<String,Object> toLTRScoringModelMap(LTRScoringModel model) {
+    final LinkedHashMap<String,Object> modelMap = new LinkedHashMap<>(5, 1.0f);
+
+    modelMap.put(NAME_KEY, model.getName());
+    modelMap.put(CLASS_KEY, model.getClass().getCanonicalName());
+    modelMap.put(STORE_KEY, model.getFeatureStoreName());
+
+    final List<Map<String,Object>> features = new ArrayList<>();
+    final List<Feature> featuresList = model.getFeatures();
+    final List<Normalizer> normsList = model.getNorms();
+    for (int ii=0; ii<featuresList.size(); ++ii) {
+      features.add(toFeatureMap(featuresList.get(ii), normsList.get(ii)));
+    }
+    modelMap.put(FEATURES_KEY, features);
+
+    modelMap.put(PARAMS_KEY, model.getParams());
+
+    return modelMap;
+  }
+
+  private static Feature featureFromFeatureMap(SolrResourceLoader solrResourceLoader,
```

Comment:
This function does a lookup to find a feature class, unlike the others which create a Normalizer/Model.  I think you should just delete this function and have the caller just call featureStore.get(featureName) directly.
Rename and remove the solrResourceLoader param if you think this is really needed